docs/ARCHITECTURE.md

```diff
@@ -67,9 +69,11 @@ EnvCheckSync -->|do exist| SshConfigureSync
 SshConfigureSync --> SetVariablesSync
 
 subgraph compareVersion["compare the sync version"]
-SetVariablesSync --> CheckTemplateFileExists
-CheckTemplateFileExists -->|exists| WriteTemplateVersionSync
-CheckTemplateFileExists -->|does not exist|CompareTemplateVersionSync
+SetVariablesSync --> CheckTemplateFileExistsGithub
```

Comment:
Thanks a lot for updating the documentation :+1: 

The rendered UML does not look right (see the pictures below).
Maybe lets make it a bit more easy and just add (for now) within the section 
`Check if the .templatesyncrc file exists in repository` and `check if .templatesyncignore file exists` that it checks first in `.github/` folder then in `root`. @LuisHenri what do you think? Then we can merge :+1: 

![image](https://user-images.githubusercontent.com/8027933/209462542-98d866c5-460a-4863-8868-d6474daf95cb.png)

![image](https://user-images.githubusercontent.com/8027933/209462547-e8ca69e5-61a6-4fa5-8c5c-bc0849321bcc.png)

Reply:
Remark: I hope that with #257 it will be a bit more easy. The hope is that at least the `.templateversionrc` file is not needed anymore

Reply:
I added it in parenthesis. What do do you think?
![image](https://user-images.githubusercontent.com/44511825/209476636-b08b53f8-f3db-48f2-abb5-90eb17d5daaf.png)

Reply:
looks good :+1:  :rocket: